### PR TITLE
App catalog

### PR DESCRIPTION
**Pull request by jayrbolton**

### CODE REVIEW

File: src/client/components/catalog/AppCatalog.tsx [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+  }
+  return (
+    <div className='b black-70 blue pointer dim' onClick={component.handleClickRuns.bind(component)}>
+      <span className='dib mr2'> Runs </span>
+      <span className={component.runsDesc ? "fa fa-caret-down" : "fa fa-caret-up"}></span>
+    </div>
+  );
+}
+
+// Loading spinner
```
Comment (FlyingUnicornSF) [inline src/client/components/catalog/AppCatalog.tsx]: do you think we should have one spinner component in the generic folder for reuse?

File: src/client/utils/fetchApps.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      version: '1.1',
+      id: '0',
+      params: [{tag: tag}]
+    })
+  });
+  const [runsResult, detailsResult] = await Promise.all([runs, details])
+  // TODO error handling cases
+  // Extract and reduce the size of the data
+  // Turn runs into an object of {id: count}
+  let runsJson = await runsResult.json();
```
Comment (FlyingUnicornSF) [inline src/client/utils/fetchApps.tsx]: is it possible for this fetch to fail and return something that you can't json or runsJson is something you can't `reduce()`? if there is that possibility,  how should it handled?

File: src/client/components/catalog/AppCatalog.tsx [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  // Fetch the app data
+  fetchData() {
+    this.setState({ loading: true });
+    fetchApps(this.tag).then((result) => {
+      const hasResults = result && result.details && result.details.length;
```
[the remark this answers is not in the document]
Comment (jayrbolton) [reply src/client/components/catalog/AppCatalog.tsx]: * hasResults is maybe an unused variable, will check :+1: 
* I don't think I got an error, but I'll check

[the remark this answers is not in the document]
Comment (jayrbolton) [reply src/client/components/catalog/AppCatalog.tsx]: Yeah I should add that

File: src/client/components/catalog/AppCatalog.tsx [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+      const data = mungeData(result);
+      this.setState({
+        rawData: data,
+        categories: result.categories,
+        loading: false,
+      });
+      this.applyResultFilters();
```
[the remark this answers is not in the document]
Comment (jayrbolton) [reply src/client/components/catalog/AppCatalog.tsx]: I think I want all filters to get reset, but I'm not seeing that happening here. I will test it

File: src/client/components/catalog/AppCatalog.tsx (hunk 1 above)
Comment (jayrbolton) [reply src/client/components/catalog/AppCatalog.tsx]: :+1: I extracted it out into components/generic/LoadingSpinner.tsx

File: src/client/components/catalog/AppCatalog.tsx [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+          { result.runs }
+        </div>
+      </div>
+    </div>
+  );
+}
+
+// Convert data from the server into objects that we use directly in the UI
+function mungeData (inpData: CombinedResult): Array<SDKApp> {
+  let ret = inpData.details.map((d: DetailsResult) => {
```
[the remark this answers is not in the document]
Comment (jayrbolton) [reply src/client/components/catalog/AppCatalog.tsx]: okay okay i changed 9it

File: src/client/utils/fetchApps.tsx (hunk 2 above)
Comment (jayrbolton) [reply src/client/utils/fetchApps.tsx]: yeah we need error handling here. I will look into adding it..

**Pull request merged**